builtin/providers/bitbucket/resource_repository.go

```diff
@@ -0,0 +1,247 @@
+	"log"
+)
+
+type CloneUrl struct {
+	Href string `json:"href,omitempty"`
+	Name string `json:"name,omitempty"`
+}
+
+type Repository struct {
+	SCM         string `json:"scm,omitempty"`
+	HasWiki     bool   `json:"has_wiki,omitempty"`
+	HasIssues   bool   `json:"has_issues,omitempty"`
+	Website     string `json:"website,omitempty"`
+	IsPrivate   bool   `json:"is_private,omitempty"`
+	ForkPolicy  string `json:"fork_policy,omitempty"`
+	Language    string `json:"language,omitempty"`
+	Description string `json:"description,omitempty"`
+	Name        string `json:"name,omitempty"`
+	UUID        string `json:"uuid,omitempty"`
+	Project     struct {
+		Key string `json:"key,omitempty"`
+	} `json:"project,omitempty"`
+	Links struct {
+		Clone []CloneUrl `json:"clone,omitempty"`
+	} `json:"links,omitempty"`
+}
+
+func resourceRepository() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceRepositoryCreate,
+		Update: resourceRepositoryUpdate,
+		Read:   resourceRepositoryRead,
+		Delete: resourceRepositoryDelete,
+		Importer: &schema.ResourceImporter{
+			State: schema.ImportStatePassthrough,
+		},
+
+		Schema: map[string]*schema.Schema{
+			"scm": &schema.Schema{
+				Type:     schema.TypeString,
+				Optional: true,
+				Default:  "git",
+			},
+			"has_wiki": &schema.Schema{
+				Type:     schema.TypeBool,
+				Optional: true,
+				Default:  false,
+			},
+			"has_issues": &schema.Schema{
+				Type:     schema.TypeBool,
+				Optional: true,
+				Default:  false,
+			},
+			"website": &schema.Schema{
+				Type:     schema.TypeString,
+				Optional: true,
+			},
+			"clone_ssh": &schema.Schema{
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"clone_https": &schema.Schema{
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"project_key": &schema.Schema{
+				Type:     schema.TypeString,
+				Optional: true,
+			},
+			"is_private": &schema.Schema{
+				Type:     schema.TypeBool,
+				Optional: true,
+				Default:  true,
+			},
+			"fork_policy": &schema.Schema{
+				Type:     schema.TypeString,
+				Optional: true,
+				Default:  "allow_forks",
+			},
+			"language": &schema.Schema{
+				Type:     schema.TypeString,
+				Optional: true,
+			},
+			"description": &schema.Schema{
+				Type:     schema.TypeString,
+				Optional: true,
+			},
+			"owner": &schema.Schema{
+				Type:     schema.TypeString,
+				Required: true,
+			},
+			"name": &schema.Schema{
+				Type:     schema.TypeString,
+				Required: true,
+			},
+		},
+	}
+}
+
+func newRepositoryFromResource(d *schema.ResourceData) *Repository {
+	repo := &Repository{
+		Name:        d.Get("name").(string),
+		Language:    d.Get("language").(string),
+		IsPrivate:   d.Get("is_private").(bool),
+		Description: d.Get("description").(string),
+		ForkPolicy:  d.Get("fork_policy").(string),
+		HasWiki:     d.Get("has_wiki").(bool),
+		HasIssues:   d.Get("has_issues").(bool),
+		SCM:         d.Get("scm").(string),
+		Website:     d.Get("website").(string),
```

Comment:
is there a chance something bad will happen if 1 of these values isn't set and passes something like an "" ?